foreman_stub.py

```diff
@@ -9,61 +9,146 @@
 from collections import defaultdict
 from urllib.parse import urlparse
 
-from flask import Flask, jsonify, request
+from flask import Flask, abort, jsonify, request
 
 
 DEBUG = os.getenv('DEBUG', 'false').lower() not in ('false', '0', '', 'off')
 
 if DEBUG:
-    import pip
-    pip.main(['install', 'q', 'epdb'])
+    try:
+        import epdb
+    except ImportError:
+        pass
+    else:
+        epdb.serve()
 
-app = Flask(__name__)
+app = Flask(__name__)  # pylint: disable=invalid-name
 
 PAGECACHE = defaultdict(dict)
+REPORTS = []
 
 
 @app.before_first_request
 def build_pagecache():
     """Read JSON fixtures into in-memory cache."""
     jfiles = glob.glob('fixtures/*.json')
     for jfile in jfiles:
-        with open(jfile, 'r') as f:
-            jdata = json.load(f)
+        with open(jfile, 'r') as fixture:
+            jdata = json.load(fixture)
         key = urlparse(jdata['url']).path[7:]
         subkey = jdata['page']
         PAGECACHE[key][subkey] = jdata['data']
 
+        if 'id' in jdata['data'] and key == f"/hosts/{jdata['data']['id']}":
+            PAGECACHE[f"/hosts/{jdata['data']['name']}"][subkey] = jdata['data']
+
 
 def get_page_num():
-    """Return page number if present."""
-    pagenum = request.args.get('page')
-    if pagenum:
-        pagenum = int(pagenum)
-    return pagenum
+    """Return page number."""
+    return int(request.args.get('page', 1))
+
+
+def get_per_page():
+    """Return the number of results per page."""
+    return int(request.args.get('per_page', 20))
+
+
+def find_host(hostid):
+    try:
+        return PAGECACHE[f'/hosts/{hostid}'][1]
+    except KeyError:
+        abort(404)
 
 
 @app.route('/api/v2/hosts')
-@app.route('/api/v2/hosts/<hostid>')
-@app.route('/api/v2/hosts/<hostid>/facts')
-def get_hosts(hostid=None):
+def get_hosts():
     """Render fixture contents from cache."""
     pagenum = get_page_num()
+    resp = PAGECACHE['/hosts'][pagenum]
+    return jsonify(resp)
 
-    cache_key = '/hosts'
-    if hostid is not None:
-        cache_key = f'{cache_key}/{hostid}'
 
-    resp = PAGECACHE[cache_key]
+@app.route('/api/v2/hosts/facts', methods=['POST'])
+def handle_facts():
```

Comment:
Please follow the tradition of adding docstrings to functions.

Reply:
It's still a draft :)